locator/tablets.cc

```diff
@@ -310,6 +310,19 @@ dht::token_range tablet_map::get_token_range(tablet_id id) const {
     }
 }
 
+dht::token_range tablet_map::get_post_split_token_range_of(const token& t) const noexcept {
+    auto get_last_token_post_split = [&] (tablet_id id) {
+        return dht::last_token_of_compaction_group(_log2_tablets + 1, size_t(id));
+    };
+
```

Comment:
let's introduce a 

get_last_token(tablet_id id, log2_tablets).

get_last_token(id) calls get_last_token(id, _log2_tablets)

then we use the former here.

Reply:
Done. I added couple of variants - `get_last_token_impl` and `get_token_range_impl` that helped with the final implementation.